### PR TITLE
Escape special characters before sending request to authnet

### PR DESCRIPTION
**Pull request by subhojit777**

### CODE REVIEW

**Comment by subhojit777** [comment]
Some history about this problem:

- https://secure.php.net/manual/en/simplexmlelement.addchild.php#112204

File: src/Request/XmlRequest.php [hunk 1]
```diff
@@ -35,7 +35,7 @@ public function arrayToXml($array, \SimpleXMLElement $xml)
                     $this->arrayToXml($value, $xml);
                 }
             } else {
-                $xml->addChild($key, $value);
+                $xml->{$key} = $value;
```
Comment (mglaman) [inline src/Request/XmlRequest.php]: @subhojit777 it'd be great to have a test explaining this change. And even linking to that comment as to why we append children that way.

**Review by mglaman — CHANGES_REQUESTED** [review]

**Comment by subhojit777** [comment]
@mglaman I have added a test to backup my changes. Please check.

**Review by mglaman — APPROVED** [review]

**Comment by mglaman** [comment]
@subhojit777 looks good. Will merge

**Comment by mglaman** [comment]
Thanks a lot!

**Pull request merged**